geode-core/src/main/java/org/apache/geode/cache/configuration/ClusterCacheElement.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.geode.cache.configuration;
+
+public interface ClusterCacheElement extends CacheElement, ServerCacheElement, LocatorCacheElement {
```

Comment:
Because the `ClusterCacheElement`'s implementation will be used on both locator/server and it holds all the configuration attributes of a CacheElement.

Reply:
I might be barking up the wrong tree, but it smacks of bad OO-design... Example... If you design a `Kennel`, and you say that there are `KennelInhabitants`, which represent the animals that live in the `Kennel`... You cannot make a `KennelInhabitant` of type `Canine` AND `Feline` just because it can be any one of the two... This is begging for failure if you have a `Canine` but you cast it to a `Feline` and try to invoke the operations on that...
Just bad OO-design... `implements` and `extends` should be replaced with "IS A..." and not with "COULD BE ONE OF THESE"